src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsTest.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Miner.Interfaces;
+using Stratis.Bitcoin.Features.Miner.Staking;
+using Stratis.Bitcoin.Features.Wallet;
+using Stratis.Bitcoin.IntegrationTests.Common;
+using Stratis.Bitcoin.IntegrationTests.Common.EnvironmentMockUpHelpers;
+using Stratis.Bitcoin.Networks;
+using Xunit;
+
+namespace Stratis.Bitcoin.IntegrationTests.Miners
+{
+    public sealed class ProofOfStakeMintCoinsTest
+    {
+        private readonly HashSet<uint256> transactionsBeforeStaking = new HashSet<uint256>();
+        private readonly ConcurrentDictionary<uint256, TransactionData> transactionLookup = new ConcurrentDictionary<uint256, TransactionData>();
+
+        [Fact]
+        public void Staking_Wallet_Can_Mint_New_Coins()
```

Comment:
Is this test actually stable?

Reply:
Yes, but it had a problem with timing out the build due to it not shutting down the node properly, There was an issue with the BDD style spec so I rewrote it, old school style.